src/datachain/lib/signal_schema.py

```diff
@@ -402,9 +402,12 @@ def _set_file_stream(
             if ModelStore.is_pydantic(finfo.annotation):
                 SignalSchema._set_file_stream(getattr(obj, field), catalog, cache)
 
-    def get_column_type(self, col_name: str) -> DataType:
+    def get_column_type(self, col_name: str, leaf_only: bool = True) -> DataType:
```

Comment:
Don't really like `leaf_only` name, if anyone have better suggestions, please, suggest 🙏

Reply:
I'll add a comment 👍

In two words: this function return type of the column if this is not a subtree column (check [here](https://github.com/iterative/datachain/pull/658/files/9cc79abc3590404f459b13a8ef2007160fb818bd#diff-69a1fc2d77ff106392ad7649ed8cf7ec344dad230af2e4d0ea9c64b816cee15bL407)). It will return `int` type for `file.size`, but will raise `SignalResolvingError` exception for `file` signal because it have subtree.

I was thinking about adding separate function, but I think it will be a code duplication :(

Reply:
`allow_subtree` instead of `leaf_only` may be?

Reply:
> is it user facing in some cases?

As far as I know users don't need it, we use it internally.

Reply:
Renamed `leaf_only` param to `with_subtree`.

Added docstring:
```python
    def get_column_type(self, col_name: str, with_subtree: bool = False) -> DataType:
        """
        Returns column type by column name.

        If `with_subtree` is True, then it will return the type of the column
        even if it has a subtree (e.g. model with nested fields), otherwise it will
        return the type of the column (standard type field, not the model).

        If column is not found, raises `SignalResolvingError`.
        """
```